soh/soh/Enhancements/mods.cpp

```diff
@@ -557,16 +559,26 @@ void UpdateMirrorModeState(int32_t sceneNum) {
     bool nextMirroredWorld = false;
 
     int16_t mirroredMode = CVarGetInteger("gMirroredWorldMode", MIRRORED_WORLD_OFF);
+    int16_t inDungeon = (sceneNum >= SCENE_YDAN && sceneNum <= SCENE_GANONTIKA_SONOGO && sceneNum != SCENE_GERUDOWAY) ||
+                        (sceneNum >= SCENE_YDAN_BOSS && sceneNum <= SCENE_GANON_FINAL) ||
+                        (sceneNum == SCENE_GANON_DEMO);
 
-    if (mirroredMode == MIRRORED_WORLD_RANDOM_SEEDED) {
-        uint32_t seed = sceneNum + (gSaveContext.n64ddFlag ? (gSaveContext.seedIcons[0] + gSaveContext.seedIcons[1] + gSaveContext.seedIcons[2] + gSaveContext.seedIcons[3] + gSaveContext.seedIcons[4]) : gSaveContext.sohStats.fileCreatedAt);
+    if (mirroredMode == MIRRORED_WORLD_RANDOM_SEEDED || mirroredMode == MIRRORED_WORLD_DUNGEONS_RANDOM_SEEDED) {
+        uint32_t seed = sceneNum + (gSaveContext.n64ddFlag ? (gSaveContext.seedIcons[0] + gSaveContext.seedIcons[1] +
+                        gSaveContext.seedIcons[2] + gSaveContext.seedIcons[3] + gSaveContext.seedIcons[4]) : gSaveContext.sohStats.fileCreatedAt);
         Random_Init(seed);
     }
 
-    uint8_t randomNumber = Random(0, 2);
+    bool randomMirror = Random(0, 2) == 1;
```

Comment:
just want to make sure i'm understanding the randomness here

in the case of seeded random, we init random with the seed and then immediately call random, meaning we should always get the same value from it

it seems this means *all* scenes/dungeons will either be mirrored or not, as opposed to the same set of scenes/dungeons being mirrored/not in that save file

am i misinterpreting this?

Reply:
The seed value in seeded mode uses `current scene + (randomizer hash icons || file start time)`
So each scene effectively as it's own seed value.

Reply:
oh, somehow i missed the `sceneNum +` when reading that, nevermind me